src/main/java/org/entando/kubernetes/model/bundle/processor/PluginProcessor.java

```diff
@@ -150,10 +150,40 @@ private void logDescriptorWarnings(PluginDescriptor descriptor) {
             log.warn(errMessage,
                     descriptor.getDockerImage(),
                     BundleUtilities.MAX_ENTANDO_K8S_POD_NAME_LENGTH,
-                    BundleUtilities.truncatePodPrefixName(deploymentBaseName));
+                    this.truncatePodPrefixName(deploymentBaseName));
         }
     }
 
+    public String composeFullDeploymentNameAndTruncateIfNeeded(PluginDescriptor descriptor, String bundleId) {
+
+        String deploymentBaseName;
+
+        if (StringUtils.hasLength(descriptor.getDeploymentBaseName())) {
+            deploymentBaseName = BundleUtilities.makeKubernetesCompatible(descriptor.getDeploymentBaseName());
+        } else {
+            deploymentBaseName = BundleUtilities.composeNameFromDockerImage(descriptor.getDockerImage());
+        }
+
+        String fullDeploymentName = deploymentBaseName + "-" + BundleUtilities.makeKubernetesCompatible(bundleId);
+
+        if (AppConfiguration.isTruncatePluginBaseNameIfLonger()) {
+            fullDeploymentName = this.truncatePodPrefixName(fullDeploymentName);
+        }
+
+        return fullDeploymentName;
+    }
+
+    protected String truncatePodPrefixName(String podPrefixName) {
+        if (podPrefixName.length() > pluginDescriptorValidator.getFullDeploymentNameMaxlength()) {
+
+            podPrefixName = podPrefixName
+                    .substring(0, Math.min(pluginDescriptorValidator.getFullDeploymentNameMaxlength(), podPrefixName.length()))
```

Comment:
@firegloves As for line 177 `podPrefixName.length()` is always greater than `pluginDescriptorValidator.getFullDeploymentNameMaxlength()`

Reply:
@w-caffiero-entando why are you saying this? the pod prefix could be also smaller than the max length, doesn't it?